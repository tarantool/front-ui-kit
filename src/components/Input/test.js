import React from 'react';
import renderer from 'react-test-renderer';
import { css, cx } from 'emotion';
import {
  Button,
  DropdownItem,
  DropdownDivider,
  IconChevron,
  IconSearch,
  Input,
  Scrollbar,
  withDropdown
} from '../../index';

jest.mock(
  '../Scrollbar/index.js',
  () => ({
    Scrollbar: ({ children, className }) => <div className={className}>{children}</div>
  })
);

const handleChange = jest.fn();
const handleClear = jest.fn();
const handleSelect = jest.fn();

const DropdownButton = withDropdown(Button);

const getDropdownOption = prefix => option => (
  <DropdownItem onClick={() => handleSelect(`${prefix}:${option.indexOf(' ') !== -1 ? `"${option}"` : option}`)}>
    {option}
  </DropdownItem>
);

const DropdownControlIcon = ({ className }) => (
  <IconChevron
    direction='down'
    className={cx(className, css`fill: rgba(245, 34, 45, 0.65);`)}
  />
);

const dropdownExample = (
  <DropdownButton
    items={(
      <Scrollbar className={css`height: 200px;`}>
        {
          [
            'Healthy',
            'Unhealthy'
          ].map(getDropdownOption('status'))
        }
        <DropdownDivider />
        {
          [
            'Stateful Connector',
            'Input_processor',
            'Connector',
            'Scheduler',
            'Storage',
            'Task_runner'
          ].map(getDropdownOption('role'))
        }
      </Scrollbar>
    )}
    intent='secondary'
    iconRight={DropdownControlIcon}
  >
    Filter
  </DropdownButton>
);

describe('Input', () => {
  it('simple', () => {
    const tree = renderer.create(
      <>
        <Input
          value={'value'}
          placeholder='Placeholder'
          onChange={handleChange}
          title='Input'
        />

        <Input
          value={'value'}
          placeholder={'Filter by uri, uuid, role, alias or labels'}
          onChange={handleChange}
          rightIcon={<IconSearch />}
        />
      </>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('with left-/rightElement', () => {
    const tree = renderer.create(
      <>
        <Input
          value={'value'}
          placeholder={'Filter by uri, uuid, role, alias or labels'}
          onChange={handleChange}
          onClearClick={handleClear}
          rightIcon={<IconSearch />}
          leftElement={dropdownExample}
        />
        <Input
          value={'value'}
          placeholder={'Filter by uri, uuid, role, alias or labels'}
          onChange={handleChange}
          rightIcon={<IconSearch />}
          rightElement={dropdownExample}
        />
      </>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
