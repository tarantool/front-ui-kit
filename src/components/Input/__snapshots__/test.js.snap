// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input simple 1`] = `
Array [
  <div
    className="css-4adaeg"
    title="Input"
  >
    <input
      className="css-1ssefzq"
      onBlur={[Function]}
      onChange={[MockFunction]}
      onFocus={[Function]}
      placeholder="Placeholder"
      value="value"
    />
  </div>,
  <div
    className="css-4adaeg"
  >
    <input
      className="css-1kkc718"
      onBlur={[Function]}
      onChange={[MockFunction]}
      onFocus={[Function]}
      placeholder="Filter by uri, uuid, role, alias or labels"
      value="value"
    />
    <div
      className="css-jgwfcl"
    >
      <svg
        className="css-p78i9h"
        viewBox="0 0 16 16"
      >
        <use
          xlinkHref="#search"
        />
      </svg>
    </div>
  </div>,
]
`;

exports[`Input with left-/rightElement 1`] = `
Array [
  <div
    className="css-1mr8eo6"
  >
    <button
      className="css-movsn2"
      onClick={[Function]}
      type="button"
    >
      Filter
      <svg
        className="css-1nldcbc"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </button>
    <input
      className="css-19jk83v"
      onBlur={[Function]}
      onChange={[MockFunction]}
      onFocus={[Function]}
      placeholder="Filter by uri, uuid, role, alias or labels"
      value="value"
    />
    <div
      className="css-jgwfcl"
    >
      <svg
        className="css-1grsr1t"
        onClick={[Function]}
        viewBox="0 0 16 16"
      >
        <use
          xlinkHref="#cancel"
        />
      </svg>
    </div>
  </div>,
  <div
    className="css-1ofuyr8"
  >
    <input
      className="css-19jk83v"
      onBlur={[Function]}
      onChange={[MockFunction]}
      onFocus={[Function]}
      placeholder="Filter by uri, uuid, role, alias or labels"
      value="value"
    />
    <div
      className="css-jgwfcl"
    >
      <svg
        className="css-p78i9h"
        viewBox="0 0 16 16"
      >
        <use
          xlinkHref="#search"
        />
      </svg>
    </div>
    <button
      className="css-movsn2"
      onClick={[Function]}
      type="button"
    >
      Filter
      <svg
        className="css-1nldcbc"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </button>
  </div>,
]
`;
